### ISSUE TITLE
Xcode 15.3 and App Store Connect Issue

### ISSUE THREAD

**Issue by johnarnaou** [facebook/facebook-ios-sdk#2353]
### Checklist before submitting a bug report

- [X] I've updated to the latest released version of the SDK
- [X] I've searched for existing [GitHub issues](https://github.com/facebook/facebook-ios-sdk/issues)
- [X] I've looked for existing answers on [Stack Overflow](https://facebook.stackoverflow.com), the [Facebook Developer Community Forum](https://developers.facebook.com/community/) and the [Facebook Developers Group](https://www.facebook.com/groups/fbdevelopers)
- [X] I've read the [Code of Conduct](https://github.com/facebook/facebook-ios-sdk/blob/main/CODE_OF_CONDUCT.md)
- [X] This issue is not security related and can safely be disclosed publicly on GitHub

### Xcode version

15.3

### Facebook iOS SDK version

14.1.0

### Dependency Manager

SPM

### SDK Framework

Core

### Goals

Archive my application and upload it to the App Store.

### Expected results

Processing of the application on the Appstore connect without issues.

### Actual results

I am getting the following errors:

<img width="1137" alt="Screenshot 2024-03-06 at 11 59 32" src="https://github.com/facebook/facebook-ios-sdk/assets/925011/a39ef3ae-ec5d-46b2-b388-e2356541fdbc">

### Steps to reproduce

1. Download Xcode 15.3
2. Add as a dependency the Facebook SDK
3. Upload a build to App Store Connect

**Comment by jadebowl** [comment]
I am also experiencing this issue with the new version of Xcode

**Comment by antoinepalazzolo** [comment]
I'm also encountering this issue and am unable to upgrade due to #2326. We urgently need attention from the Facebook team on this matter.

**Comment by sam-bensley** [comment]
I was able to get around this issue temporarily by downgrading my Xcode to 15.0.1 via the following https://developer.apple.com/download/applications/
Not an idea scenario but at least i can push to app store connect now.

**Comment by neobeppe** [comment]
+1, not possible to update apps with FB right now

**Comment by mmdumi** [comment]
+1

**Comment by larryonoff** [comment]
+1

**Comment by alpha-moisol** [comment]
+1

**Comment by SteffenKeller** [comment]
+1

**Comment by jondlew** [comment]
+1

**Comment by HSCOGT** [comment]
I'm also experiencing this issue with the new version of Xcode.

**Comment by gooson** [comment]
+1

**Comment by martarodriguezm** [comment]
+1

**Comment by alpha-moisol** [comment]
Dear Facebook Team, we really do need this fixed! This issue is almost 3 days old

**Comment by michalnowak061** [comment]
same here

**Comment by johnarnaou** [comment]
Is this repo active ? Last update was 3 weeks ago.. I see multiple issues that are not closed or triaged from a long time ago. Is this the correct place to report this issue ? Does anyone know ?

**Comment by neobeppe** [comment]
@johnarnaou I also sent an email to the dev that made last tag 3 month ago. I saw more commits and more pending releases like 17.0.0

**Comment by neobeppe** [comment]
Got an answer from FB devs:

> Hi Giuseppe,
> 
> Thanks for reporting the issue. We're now working on that and will keep the community updated.
> 
> Best,
> Zilin

**Comment by al3x88** [comment]
I have the same issue :( I am desperate! If you guys have some workaround while FB support release a fix .........

**Comment by Arideno** [comment]
> I have the same issue :( I am desperate! If you guys have some workaround while FB support release a fix .........

Just install previous Xcode, you can publish to AppStore Connect with it

**Comment by hanlinag** [comment]
+1

**Comment by ersel95** [comment]
Having the same issue with Xcode version 15.3. Facebook and Google had the same issue and solved with an update. Waiting for an update from FB ASAP.

**Comment by Ariandr** [comment]
+1

**Comment by 0xWOF** [comment]
Hello, I experience same issue when I write iOS Framework. And I solve this issue with updating iOS Framework's Info.plist's MinimumOSVersion from 11.0 to 12.0. I think the issue can be solved by update Facebook iOS SDK to 15+. (14's MinimumOSVersion is 11.0 and 15's MinimumOSVersion is 12.0) If you can not update Facebook iOS SDK, you can solve the issue with temporary modifying Facebook iOS SDK's xcframework's framework's Info.plist's MinimumOSVersion 11.0 to 12.0. (Personally I do not recommend this way)

**Comment by ed-smokefree-ai** [comment]
+1

Have had to change my Xcode Cloud workflow to use Xcode 15.2 as a temporary workaround until this is fixed.

**Comment by VadimYakovliev** [comment]
I have the same issue

**Comment by ersel95** [comment]
You probably installed the SPM with Dependency Rule which is limiting the version. I changed Facebook-Ios-Sdk Dependency rule to main Branch and everything is good to go now.

![Screenshot 2024-03-11 at 15 56 44](https://github.com/facebook/facebook-ios-sdk/assets/11449481/6afd5765-142c-41a3-b3da-25adf85f4d85)

**Comment by VadimYakovliev** [comment]
> I changed Facebook-Ios-Sdk Dependency rule to main Branch and everything is good to go now.

Nothing has changed, same error.

**Comment by umutturemen1496** [comment]
Hello Dear Facebook Team, please fix the issue as soon as possible. I am using the main branch to get the most updated version as soon as possible. Nothing has changed since 3 days. Either Apple or you should fix this quickly. We cannot send an update. 

![Screenshot 2024-03-11 at 14 18 48](https://github.com/facebook/facebook-ios-sdk/assets/83616491/e5f6c756-304a-4590-90f0-3088bc469fd5)

**Comment by krotoff** [comment]
+1 here, waiting for the update

**Comment by jadebowl** [comment]
If you need a workaround you could point the Package to the commit **e347ccea13bae0e03f171a8d1840a8bc9778264f**, they forgot to add the tag but they already bumped to version 17, 3 weeks ago. It worked for me, I was able to Archive my build and send it to App Store Connect.

**Comment by ostatnicky** [comment]
> If you need a workaround you could point the Package to the commit **[e347cce](https://github.com/facebook/facebook-ios-sdk/commit/e347ccea13bae0e03f171a8d1840a8bc9778264f)**. They forgot to add the tag but they already bumped to version 17.0.0 3 weeks ago. It worked for me, I was able to Archive my build and send it to App Store Connect. Could be a temporary solution if you need to send it asap.

No, it doesn't work.

**Comment by dppeak** [comment]
+1

**Comment by sharva89** [comment]
+1

**Comment by SamPetherbridge** [comment]
+1

**Comment by neriusv** [comment]
+1

**Comment by levochkaa** [comment]
Same issue here, unable to move to Xcode 15.3, because of that. Latest commit doesn't work, too

**Comment by ExploriaTeam** [comment]
+1

**Comment by SajjadKharrazi** [comment]
+1

**Comment by rajlunia** [comment]
This is happening everywhere.
This is the only useful information I could find, no clue how they came up with that temp workaround, though. https://github.com/googleads/swift-package-manager-google-mobile-ads/issues/62#issuecomment-1981219533

Xcode 15.2 (works), and 15.3 (fails)

Re-tooling commits branching platform for path differences https://github.com/firebase/firebase-ios-sdk/pull/12517/files
https://github.com/facebook/facebook-ios-sdk/issues/2353
https://github.com/google/GoogleAppMeasurement/issues/62
etc, etc.

**Comment by LaurentiuUngur** [comment]
Any updates on this ?

**Comment by Jordan3900** [comment]
+1

**Comment by KimHyunMin1751** [comment]
+1

**Comment by oleksandryevdokymov** [comment]
I have the same issue. Is there any updates on this?

**Comment by michalnowak061** [comment]
@ryantobinmeta any updates?

**Comment by samcolak** [comment]
+1

**Comment by samcolak** [comment]
> Seems like Apple problem

Normally I'd agree but knowing Apple (as I do), they will point fingers at people not following the (amended) specification - lol

**Comment by mneuburg-livefront** [comment]
+1

**Comment by RinStar2003** [comment]
+1

**Comment by rromanchuk** [comment]
Does the ads team even know this is their product? Or did they just give up after ATT?  Just fast follow other ad networks, at least on the communication part. Deflect to apple, whatever, but bad vibes letting this rot here. 

TLDR: (+1 xoxoxo)

**Comment by samcolak** [comment]
> Does the ads team even know this is their product? Or did they just give up after ATT? Just fast follow other ad networks, at least on the communication part. Deflect to apple, whatever, but bad vibes letting this rot here.
> 
> TLDR: (+1 xoxoxo)

I imagine that they either 1) care very little about integration partners or 2) this isn't in their scope - In so far as I can understand this happens to be related to the build components of the framework more than the code itself. It can be that the adage, "if it aint broke, dont fix it" applies - For the last 4-5 hours ive been looking to where this "MinimumOSVersion" should be placed - frankly ive been through alot of combinations - If you are using SPM (Swift Package Manager), dropping back to 15.1 (on Xcode Cloud or local building) seems to have some benefits (as long as you have the package.resolved file appearing correctly) - If you are running ObjectiveC (with a bridge) into FB its a bigger headache to drop down to 15.1 - It just dont play ball..

BTW if anyone didn't notice, in 15.3 to have a repo appearing in the GitHub repos, simply "Star" the repo and refresh for it to appear

Take care all !

**Comment by samcolak** [comment]
Oh - side note - To have a repo appearing in SPM (under repos)....

**Comment by johnarnaou** [comment]
I hope this works 🤞 

![IMG_6541](https://github.com/facebook/facebook-ios-sdk/assets/925011/c1724f92-e581-4303-97ed-5cb2a47cca1e)

**Comment by samcolak** [comment]
SOLVED !!

**Comment by RinStar2003** [comment]
> SOLVED !!

How

**Comment by samcolak** [comment]
Solution requires a few steps - Please bare with me..

1. Download the Framework files (at https://github.com/facebook/facebook-ios-sdk/releases for 16.3.1)
2. Extract the zips
3. Download a copy of the Facebook-ios-sdk
4. Modify the file "package.swift" - commenting lines 245 - 249 EXCEPT line 246 (self=.local)
5. Copy the extracted zip files into the Facebook folder under "build/XCFrameworks/Static"
6. In each framework zip modify the file "Info.plist" under "ios-arm64" AND 'iOS-arm64_x86_64-simulator" - You'll see MinimumOSVersion = 12.0 / Change this to 100.0
7. Remove the dependancies in your project (linking to the Facebook sdk @ GitHub)
8. Add dependancy to local instance

DONE !!

**Comment by martarodriguezm** [comment]
> Solution requires a few steps - Please bare with me..
> 
> 1. Download the Framework files (at https://github.com/facebook/facebook-ios-sdk/releases for 16.3.1)
> 2. Extract the zips
> 3. Download a copy of the Facebook-ios-sdk
> 4. Modify the file "package.swift" - commenting lines 245 - 249 EXCEPT line 246 (self=.local)
> 5. Copy the extracted zip files into the Facebook folder under "build/XCFrameworks/Static"
> 6. In each framework zip modify the file "Info.plist" under "ios-arm64" AND 'iOS-arm64_x86_64-simulator" - You'll see MinimumOSVersion = 12.0 / Change this to 100.0
> 7. Remove the dependancies in your project (linking to the Facebook sdk @ GitHub)
> 8. Add dependancy to local instance
> 
> DONE !!

So it is not solved, you just found a workaround.
I finally downgrade xcode to 15.2 until Facebook or Apple solve the problem.

**Comment by RinStar2003** [comment]
> So it is not solved, you just found a workaround. I finally downgrade xcode to 15.2 until Facebook or Apple solve the problem.

When I downgraded, I see this in my projects: Package.resolved file is corrupted or malformed; fix or delete the file to continue: unknown 'PinsStorage' version '3' at ... 

How to fix this?

**Comment by alpha-moisol** [comment]
> > So it is not solved, you just found a workaround. I finally downgrade xcode to 15.2 until Facebook or Apple solve the problem.
> 
> When I downgraded, I see this in my projects: Package.resolved file is corrupted or malformed; fix or delete the file to continue: unknown 'PinsStorage' version '3' at ...
> 
> How to fix this?

Delete Package.resolved file and open project in Xcode 15.2

**Comment by samcolak** [comment]
Per the above (in step 6) - theres a directory (usually the name of the sdk framework file) before you see the Info.plist

You need to do the 2 Info files (in simulator and iOS) as the CI infrastructure runs to check this on the GitHub repo

Additional note - No its not solved @martarodriguezm - Its a workaround to move us all forward for now

Per anyone downgrading, you'll see an issue with Package.resolved - this doesnt get autogenerated by the SPM at the moment - seems theres another bug...

**Comment by samcolak** [comment]
> > Solution requires a few steps - Please bare with me..
> > 
> > 1. Download the Framework files (at https://github.com/facebook/facebook-ios-sdk/releases for 16.3.1)
> > 2. Extract the zips
> > 3. Download a copy of the Facebook-ios-sdk
> > 4. Modify the file "package.swift" - commenting lines 245 - 249 EXCEPT line 246 (self=.local)
> > 5. Copy the extracted zip files into the Facebook folder under "build/XCFrameworks/Static"
> > 6. In each framework zip modify the file "Info.plist" under "ios-arm64" AND 'iOS-arm64_x86_64-simulator" - You'll see MinimumOSVersion = 12.0 / Change this to 100.0
> > 7. Remove the dependancies in your project (linking to the Facebook sdk @ GitHub)
> > 8. Add dependancy to local instance
> > 
> > DONE !!
> 
> So it is not solved, you just found a workaround. I finally downgrade xcode to 15.2 until Facebook or Apple solve the problem.

Indeed - its an intermediate fix

**Comment by samcolak** [comment]
The way the Facebook-sdk operates (for iOS) is that it downloads the framework files from GitHub - Theres no way around this than for FB to update the framework files directly. Not sure thats going to happen short-term

**Comment by martarodriguezm** [comment]
For those downgrading as I did, just remove all the dependencies and import them again, it work for me, I have already my update live in the store.

**Comment by LaurentiuUngur** [comment]
Found another solution by integrating it using Cocoapods

**Comment by andreyoshev** [comment]
+1

**Comment by Dimillian** [comment]
Any update?????????

**Comment by RinStar2003** [comment]
One of the basic and simplest solutions is to download Xcode 15.2 and use it for now

**Comment by cathalfarrell** [comment]
I can't submit to the App Store using Xcode 15.3 until this is fixed so I've had to revert to 15.2 instead.

**Comment by NicolaiBure** [comment]
Have to delete the package, and comment usage. Next week Im going to try again. You need to test when Xcode and iOS updated or before in betas

**Comment by ersel95** [comment]
> Have to delete the package, and comment usage. Next week Im going to try again. You need to test when Xcode and iOS updated or before in betas

You can use `#if canImport(Module)` rather than commenting the code. With this way, you do not have to comment & uncomment.

**Comment by neobeppe** [comment]
Don't know if devs should stop using FB or FB should change devs

**Comment by mkandler** [comment]
+1

**Comment by rromanchuk** [comment]
> 1) care very little about integration partners 

Don't hurt TEMU's feelings like that 😆

**Comment by ottosen** [comment]
Come on now Facebook! This ought to be dead easy to fix.
It is very tempting to just bin this package and remove FB support.

**Comment by Wei18** [comment]
Another workaround via downgraded to Xcode 15.2 
```sh
#!/bin/sh
mint run XcodesOrg/xcodes install --select 15.2 
```
ref:
- https://github.com/yonaskolb/Mint
- https://github.com/XcodesOrg/xcodes

**Comment by omz1990** [comment]
+1

**Comment by ArturoLee** [comment]
+1

**Comment by SajjadKharrazi** [comment]
any update!?!?!?

**Comment by vahidsayad** [comment]
It's 2 weeks and there is no fix yet.
What the support!

It's temp fix that works for me:

I downloaded following frameworks: (I didn't use rest of FB frameworks)
```
FBSDKCoreKit_Basics
FBSDKCoreKit
FBAEMKit
```

and modified this edit:

```
In each framework zip modify the file "Info.plist" under "ios-arm64" - You'll see MinimumOSVersion = 12.0 / Change this to 100.0
```

My building and submission to AppStore was success.

(credit to [samcolak](https://github.com/samcolak))

**Comment by Jodebjo** [comment]
+1

**Comment by davut** [comment]
Can we at least get estimate time this task is going to be done?

**Comment by johnarnaou** [comment]
> Can we at least get estimate time this task is going to be done?

Sure @davut 

Here are the steps towards your answer:

1. Wait until someone opens the repo and checks the issues

2. Trust that this person will contact someone at Meta. Probably their manager.

3. If the manager is technical they will probably let their know manager about the issue.

4. If their manager is technical enough then maybe this issue will be prioritized for the next quarter goals and under the maintenance KPI.

*IMPORTANT:* Repeat steps 1-4 until you get to the point this issue actually gets prioritized for next quarter.

5. Once they bring this to the next quarter, the team needs to decide if it fits in their sprint.

6. If there are more important product goals (e.g. change color of label) then this will be probably be pushed for next sprint and so on.

7. Issue gets into the sprint and you get an estimate of the date that this issue will be resolved.

**Comment by krishan226** [comment]
Hi everyone,
I finally managed to fix the persistent issue I've been facing across all my projects for the past 3 days. For those experiencing similar challenges, here's the solution that worked for me:

* Downgrade Xcode Version: Initially, I was using the latest Xcode version (15.3), but I found that downgrading to an older version resolved the problem. I uninstalled Xcode 15.3 and installed version 15.1 instead.
 * You can download older versions of Xcode from [this link](https://xcodereleases.com/).
* Resolve Package Dependency Issue: After downgrading Xcode, I addressed any package dependency issues that arose as a result of the version change.
*  If you encounter package dependency issues with your SDK packages, you can find helpful guidance on resolving them [here](https://stackoverflow.com/questions/67185817/package-resolved-file-is-corrupted-or-malformed).
I hope this solution proves helpful for others facing similar challenges.
Best regards,
Krishan Pahadiya

**Comment by devedup** [comment]
Suggesting to use Xcode 15.2 is not a solution if you're building and deploying via Xcode Cloud.

**Comment by ed-smokefree-ai** [comment]
> Suggesting to use Xcode 15.2 is not a solution if you're building and deploying via Xcode Cloud.

@devedup it is a temporary solution, just modify your Xcode Cloud workflow to use `Xcode 15.2` instead of `Latest Release`.

<img width="1014" alt="Screenshot 2024-03-19 at 12 51 05" src="https://github.com/facebook/facebook-ios-sdk/assets/113444160/3bb5bdf8-42a9-4d07-bf73-b5174d4fadb3">

**Comment by devedup** [comment]
> > Suggesting to use Xcode 15.2 is not a solution if you're building and deploying via Xcode Cloud.
> 
> @devedup it is a temporary solution, just modify your Xcode Cloud workflow to use `Xcode 15.2` instead of `Latest Release`.


Cool, didn't know this was an option. Thanks

**Comment by ozgurshn** [comment]
Same issue with Xcode 15.3 Facebook SDK 16.3.1

**Comment by tnorbert** [comment]
I tried every suggested fix but only **downgrading to xcode 15.2** helped :( This is madness that weeks passed since this bug revealed itself and other devs already fixed their library (like firebase)

**Comment by EverGod** [comment]
This is a serious showstopper. Why is this opened for so long without a solution?

**Comment by colemancda** [comment]
Also seeing this in my apps.
<img width="697" alt="Screenshot 2024-03-20 at 11 07 45 PM" src="https://github.com/facebook/facebook-ios-sdk/assets/3419766/5cc5f393-336f-4091-bb80-0500a7623c7b">

**Comment by sashastg** [comment]
From the organizer -> open the archived by finder -> open the content of the app -> frameworks -> framework which has the issue -> info.plist

MinimumOsVersion = 15.0.0

**Comment by arthurofbabylon** [comment]
> From the organizer -> open the archived by finder -> open the content of the app -> frameworks -> framework which has the issue -> info.plist
> 
> MinimumOsVersion = 15.0.0

This works. 

Note that the Minimum OS Version should match the app's Minimum OS Version – not necessarily 15.0.0. 

Note that you'll need to do some right-clicking > Open Package Contents in order to find the frameworks' info.plist file. 

Thanks @sashastg

**Comment by cemege** [comment]
Same issue:
- Xcode 15.3 (15E204a)
- Facebook SDK 16.3.1

**Comment by srostagno** [comment]
Hello, same issue, had to rollback to Xcode 15.2 :(

This didn't work for me:

> > From the organizer -> open the archived by finder -> open the content of the app -> frameworks -> framework which has the issue -> info.plist
> > MinimumOsVersion = 15.0.0
> 
> This works.
> 
> Note that the Minimum OS Version should match the app's Minimum OS Version – not necessarily 15.0.0.
> 
> Note that you'll need to do some right-clicking > Open Package Contents in order to find the frameworks' info.plist file.
> 
> Thanks @sashastg

**Comment by AntonPetrov83** [comment]
I have Xcode 15.0.1 and similar error is there. As I can see it is an error reported by App Store Connect after remote validation of the bundle. I have tried altool and Xcode Organizer to upload the build - outcome is the same:
```
Error: Asset validation failed Invalid Bundle. The bundle ***.app/Frameworks/IASDKCore.framework does not support the minimum OS Version specified in the Info.plist.
```

Very intriguing that XCode 15.2 solves the issue of the App Store Connect...

**Comment by streem-ua** [comment]
Same here!

```
Please correct the following issues and upload a new binary to App Store Connect.

ITMS-90208: Invalid Bundle - The bundle ***.app/Frameworks/FBAEMKit.framework does not support the minimum OS Version specified in the Info.plist.

ITMS-90208: Invalid Bundle - The bundle ***.app/Frameworks/FBSDKCoreKit.framework does not support the minimum OS Version specified in the Info.plist.

ITMS-90208: Invalid Bundle - The bundle ***.app/Frameworks/FBSDKCoreKit_Basics.framework does not support the minimum OS Version specified in the Info.plist.
```

<img width="747" alt="Screenshot 2024-03-21 at 19 42 07" src="https://github.com/facebook/facebook-ios-sdk/assets/3996453/c2c1277c-aa24-4031-a88d-02ae9c2751b5">

**Comment by Rukh** [comment]
The fact that the bug has not closed for 3 weeks is, of course, a shame 🥸

My hack for this. Just Add the Build Phase with the script:

```
TARGET_MIN_OS_VERSION="15.0"
FRAMEWORKS_DIR="${BUILT_PRODUCTS_DIR}/${FRAMEWORKS_FOLDER_PATH}"

update_framework_plists() {
  local path="$1"
  echo "Updating Info.plist files in $path"
  
  find "$path" -name Info.plist | while read plist; do
    local normalized_plist_path=$(realpath "$plist")
    
    echo "Updating $normalized_plist_path"
    /usr/libexec/PlistBuddy -c "Add :MinimumOSVersion string $TARGET_MIN_OS_VERSION" "$plist" &>/dev/null || /usr/libexec/PlistBuddy -c "Set :MinimumOSVersion $TARGET_MIN_OS_VERSION" "$plist"
    
    if [ $? -eq 0 ]; then
      echo "Successfully set MinimumOSVersion to $TARGET_MIN_OS_VERSION in $normalized_plist_path"
    else
      echo "Failed to set MinimumOSVersion in $normalized_plist_path"
    fi
  done
}

update_framework_plists "$FRAMEWORKS_DIR"

```

**Comment by rromanchuk** [comment]
[I hope llama tokenizes this thread](https://github.com/meta-llama/llama/commit/54c22c0d63a3f3c9e77f43a6a3041c00018f4964)

**Comment by Maxim-Inv** [comment]
I had tried to resolve this issue and I created ci_post_xcodebuild.sh with these content:
```
#!/bin/sh
set -e
if [[ -n $CI_ARCHIVE_PATH ]];
then
    # Update minimumOSVersion
    cd $CI_ARCHIVE_PATH/Products/Applications/$CI_XCODE_SCHEME.app/Frameworks/

    files=("FBAEMKit.framework/Info.plist" "FBSDKCoreKit_Basics.framework/Info.plist" "FBSDKCoreKit.framework/Info.plist")

    # Loop through each file
    for file in "${files[@]}"
    do
        plutil -convert xml1 "$file"
        sed -i '' 's/<string>12.0<\/string>/<string>100.0<\/string>/g' "$file"
        plutil -convert binary1 "$file"
    done
else
    echo "Archive path isn't available."
fi
```
When I tested it on locally created archive (use separated script file with files and loop only) it works well and changes minimusOSVersion up to 100.0. And then I can successfully upload my app to the TestFlight. 

But it doesn't affect to XCode Cloud. That is, I see that the script processed after the building but XCode Cloud still encounters this error. It seems that it collects artifacts from another folder. 

Does anyone know where XCode Cloud gets its artifacts from?

**Comment by serdar-coskun** [comment]
I have the same issue. 
Xcode 15.3

**Comment by dreampowder** [comment]
I have the same issue. 

Xcode 15.3

**Comment by dreampowder** [comment]
> Here solution:
> 
> From the organizer -> open the archived by finder -> open the content of the app -> frameworks -> framework which has the issue -> info.plist
> 
> MinimumOsVersion = 15.0.0

This solved my issue with Xcode 15.3

**Comment by umair-khan97** [comment]
> > Here solution:
> > From the organizer -> open the archived by finder -> open the content of the app -> frameworks -> framework which has the issue -> info.plist
> > MinimumOsVersion = 15.0.0
> 
> This solved my issue with Xcode 15.3

Thanks

**Comment by Maxim-Inv** [comment]
> The fact that the bug has not closed for 3 weeks is, of course, a shame 🥸
> 
> My hack for this. Just add the Build Phase with the script:

Thanks. But it's better to set the minimumOSVersion to 100.0 for all occasions.

**Comment by Rukh** [comment]
> > The fact that the bug has not closed for 3 weeks is, of course, a shame 🥸
> > My hack for this. Just add the Build Phase with the script:
> 
> Thanks. But it's better to set the minimumOSVersion to 100.0 for all occasions.
> 
> `TARGET_MIN_OS_VERSION="100.0"`

Why 100? Why not 1000 or 12345? This magic number makes no sense 🙂. In my case the problem is with the SPM package, I don’t have stable info.plist in this case

**Comment by Maxim-Inv** [comment]
> > Thanks. But it's better to set the minimumOSVersion to 100.0 for all occasions.
> > `TARGET_MIN_OS_VERSION="100.0"`
> 
> Why 100? Why not 1000 or 12345? This magic number makes no sense 🙂. And in my case, the problem is with the SPM package. I don’t have stable info.plist in this case and I can not change this property manually

I saw this number in the Google SDK. You can choose any other number, 20, 30, etc. My project minimum deployment iOS version is equal to 16.0, so this leads to an error. Soon we can change it to 17.0, and then we have to change TARGET_MIN_OS_VERSION too. To avoid this, I set purposely the large number.

P.S. I'm not sure that this issue will be fixed soon :(

**Comment by jasval** [comment]
+1 (Nasty)

**Comment by wleivsco** [comment]
+1 please fix this ASAP.

**Comment by nickx3389** [comment]
Any update here? It's been 3 weeks since this issue was posted!

**Comment by ottosen** [comment]
This is pathetic!

**Comment by anton-alekseev-amo** [comment]
+1
Please, fix this ASAP

**Comment by ostatnicky** [comment]
It looks like it's time to think about implementing Facebook Login without Facebook SDK.

https://developers.facebook.com/docs/facebook-login/guides/advanced/manual-flow
https://dev.to/marinbenc/implementing-facebook-login-on-ios-without-facebook-sdk-3k05
https://github.com/facebook/facebook-ios-sdk/issues/1437

**Comment by chrlyck** [comment]
> Here solution:
> 
> From the organizer -> open the archived by finder -> open the content of the app -> frameworks -> framework which has the issue -> info.plist (double click, to open in xcode, if you open in editor its looks like encoded, but in xcode its ok)
> 
> MinimumOsVersion = 15.0.0

Works for me - except I have to set it to 16.0 since that's the minimum target of my app.

**Comment by strejdashek** [comment]
This is really poor SDK support. Such problems should be solved in days, not weeks or months.

**Comment by vittorioscocca** [comment]
> > Here solution:
> > From the organizer -> open the archived by finder -> open the content of the app -> frameworks -> framework which has the issue -> info.plist
> > MinimumOsVersion = 15.0.0
> 
> This solved my issue with Xcode 15.3

Thanks,  this is a great solution!

**Comment by EverGod** [comment]
Hi all and thank you for posting various solutions. 
But nothing is a real solution vs the Facebook team to just add a few MinimumOsVersion key and values to some plists.

I will give this a few more days until removing the SDK, due to lack of support.

**Comment by pepejeria** [comment]
> Hi all and thank you for posting various solutions. But nothing is a real solution vs the Facebook team to just add a few MinimumOsVersion key and values to some plists.
> 
> I will give this a few more days until removing the SDK, due to lack of support.

There has not been one single reply from the Facebook team yet on this issue. This clearly show that they don't really care, so why should we. I will also be working on removing this framework that has only given me stress.

**Comment by alpha-moisol** [comment]
It's a cringe. We cannot remove this framework because of Facebook Ads Campaigns. Meta as always...

**Comment by rycbn** [comment]
+ 1

**Comment by fury21** [comment]
+++1

**Comment by Maxim-Inv** [comment]
@EverGod, @pepejeria, @alpha-moisol use [this script](https://github.com/facebook/facebook-ios-sdk/issues/2353#issuecomment-2013827793)
It does what Meta can't do for so long.

**Comment by johnarnaou** [comment]
I pinged him again. Let's see.

![image](https://github.com/facebook/facebook-ios-sdk/assets/925011/cdc333a2-5ddb-495d-91b7-0b9c0cb57c8a)

**Comment by ottosen** [comment]
Seems to be fixed with version 17.0.0 released today 🎉

**Comment by michalnowak061** [comment]
Can anyone confirm that the latest release fixes this problem?

**Comment by Ferdens** [comment]
> Can anyone confirm that the latest release fixes this problem?

@michalnowak061 

Just uploaded a build with the latest release. Everything is fine so far

**Comment by michalnowak061** [comment]
> > Can anyone confirm that the latest release fixes this problem?
> 
> @michalnowak061
> 
> Just uploaded a build with the latest release. Everything is fine so far

Nice. Thank's

**Comment by johnarnaou** [comment]
It works. Thanks everyone

**Comment by nomimalik1769** [comment]
Not working with 14.1.0

**Comment by mbuchetics** [comment]
> Not working with 14.1.0

You need to use the latest version (17.0)

**Comment by bekroz** [comment]
> I was able to get around this issue temporarily by downgrading my Xcode to 15.0.1 via the following https://developer.apple.com/download/applications/ Not an idea scenario but at least i can push to app store connect now.

I've tried that, but it's failing inside ASC. Showing "its' invalid binary".

**Comment by gun-bro98** [comment]
I also had this problem and tried to install with SPM, but there was no suitable version.
I figured it out after several attempts.

Please change the Facebook version in Package Dependencies of the project.

![스크린샷 2024-04-11 오후 1 17 26](https://github.com/facebook/facebook-ios-sdk/assets/64679046/5bb6d701-be71-4ada-8b3d-6488ff25db3a)

**Comment by bekroz** [comment]
> I also had this problem and tried to install with SPM, but there was no suitable version. I figured it out after several attempts.
> 
> Please change the Facebook version in Package Dependencies of the project.
> 
> ![스크린샷 2024-04-11 오후 1 17 26](https://private-user-images.githubusercontent.com/64679046/321501715-5bb6d701-be71-4ada-8b3d-6488ff25db3a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI4MTQ3NzcsIm5iZiI6MTcxMjgxNDQ3NywicGF0aCI6Ii82NDY3OTA0Ni8zMjE1MDE3MTUtNWJiNmQ3MDEtYmU3MS00YWRhLThiM2QtNjQ4OGZmMjVkYjNhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDExVDA1NDc1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg4MDdiMDM1MDgxM2MzYzVmNmRmNGU1Y2JlOGVlODJiMGI2ZTliNTJmZmFlN2UwYjNiMjQwOWNmM2IwYTAyNmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ufrkH9wB3ixTAGLXJjhjl-ZTV64uS0m2LU9MtDpMZfw)

This one is working solution. Thank, @gun-bro98. You saved my day!

**Comment by naticio** [comment]
how do you upgrade to 17 version? my package dependency is grayed out
<img width="787" alt="image" src="https://github.com/user-attachments/assets/8684ae59-affb-4713-b639-5fc9942ffd86">
